site/src/modules/resources/AgentDevcontainerCard.tsx

```diff
@@ -0,0 +1,73 @@
+import Link from "@mui/material/Link";
+import type { Workspace, WorkspaceAgentDevcontainer } from "api/typesGenerated";
+import { ExternalLinkIcon } from "lucide-react";
+import type { FC } from "react";
+import { portForwardURL } from "utils/portForward";
+import { AgentButton } from "./AgentButton";
+import { AgentDevcontainerSSHButton } from "./SSHButton/SSHButton";
+import { TerminalLink } from "./TerminalLink/TerminalLink";
+
+type AgentDevcontainerCardProps = {
+	container: WorkspaceAgentDevcontainer;
+	workspace: Workspace;
+	host: string;
+	agentName: string;
+};
+
+export const AgentDevcontainerCard: FC<AgentDevcontainerCardProps> = ({
+	container,
+	workspace,
+	agentName,
+	host,
+}) => {
+	return (
+		<section
+			className="border border-border border-dashed rounded p-6 "
+			key={container.id}
+		>
+			<header className="flex justify-between">
+				<h3 className="m-0 text-xs font-medium text-content-secondary">
+					{container.name}
+				</h3>
+
+				<AgentDevcontainerSSHButton
+					workspace={workspace.name}
+					container={container.name}
+				/>
+			</header>
+
+			<h4 className="m-0 text-xl font-semibold">Forwarded ports</h4>
+
+			<div className="flex gap-4 flex-wrap mt-4">
+				<TerminalLink
+					workspaceName={workspace.name}
+					agentName={agentName}
+					containerName={container.name}
+					userName={workspace.owner_name}
+				/>
+				{container.ports.map((port) => {
+					return (
+						<Link
+							key={port.port}
+							color="inherit"
+							component={AgentButton}
+							underline="none"
+							startIcon={<ExternalLinkIcon className="size-icon-sm" />}
+							href={portForwardURL(
+								host,
+								port.port,
+								agentName,
+								workspace.name,
+								workspace.owner_name,
+								location.protocol === "https" ? "https" : "http",
```

Comment:
@johnstcn should we return the port protocol in the response? Here, I'm just using whatever the user is currently using in the UI but maybe, some apps use HTTP instead of HTTPS... I'm not sure if it makes sense tho 🤔

Reply:
> I think it's always going to be the same protocol as the Coder deployment?

I'm not sure but would be safest to rely in the devcontainer config.